focus-android/app/src/main/java/org/mozilla/focus/widget/TelemetrySwitchPreference.kt

```diff
@@ -32,6 +32,9 @@ internal class TelemetrySwitchPreference(context: Context, attrs: AttributeSet?)
             .setUploadEnabled(isChecked).isCollectionEnabled = isChecked
 
         Glean.setUploadEnabled(isChecked)
+        if (isChecked) {
+            TelemetryHolder.get().recordSessionStart()
+        }
```

Comment:
Could you help me to clarify what is the context around the fix? It's not straightforward how crash was generated and the fix address it, thanks in advance!

Reply:
Because Telemetry is not activated TelemetryHolder.get().recordSessionStart() doesn't get call like it should in onresume(inside the method are some ifs and it doesn't work like it should). After the user activates the telemetry and  leaves the app TelemetryHolder.get().recordSessionStop() gets called and it crashes because TelemetryHolder.get().recordSessionStart() didn't get called .

Reply:
Thanks for the clarification!